redis/src/connection.rs

```diff
@@ -882,7 +990,33 @@ pub(crate) fn create_rustls_config(
                 })?
         } else {
             config_builder.with_no_client_auth()
-        }
+        };
+
+        // The cfg here matches the cfg on `danger_accept_invalid_hostnames`.
+        // This way adding `tls-rustls` on top of `tls-native-tls` does not
+        // remove the field or cause it to be silently ignored.
+        #[cfg(any(feature = "tls-rustls-insecure", feature = "tls-native-tls"))]
```

Comment:
since this is scoped under `#[cfg(feature = "tls-rustls")]` (line 954), the `feature = "tls-native-tls"` part should be dropped. This function is only for rustls, and you wouldn't want to accidentally enable it by setting `tls-native-tls`

Reply:
Just to make totally sure you can merge this either way, I did two things

- change the existing code to make it clearer what's going on (now included in this PR)
- also make a separate commit that deletes the code (jorendorff/redis-rs#1)

I think you can merge that commit into this PR by doing:

```
git remote add jorendorff https://github.com/jorendorff/redis-rs.git
git fetch jorendorff accept-invalid-hostnames 
git checkout accept-invalid-hostnames
git fetch jorendorff simplify-invalid-hostnames
git cherry-pick jorendorff/simplify-invalid-hostnames
git push jorendorff accept-invalid-hostnames
```